node/pkg/p2p/watermark_test.go

```diff
@@ -0,0 +1,192 @@
+package p2p
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"fmt"
+	"github.com/certusone/wormhole/node/pkg/accountant"
+	node_common "github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/governor"
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	"github.com/certusone/wormhole/node/pkg/supervisor"
+	"github.com/ethereum/go-ethereum/crypto"
+	p2pcrypto "github.com/libp2p/go-libp2p-core/crypto"
+	p2ppeer "github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p/p2p/net/connmgr"
+	"go.uber.org/zap"
+	"testing"
+	"time"
+)
+
+type G struct {
+	// arguments passed to p2p.New
+	obsvC                  chan *gossipv1.SignedObservation
+	obsvReqC               chan *gossipv1.ObservationRequest
+	obsvReqSendC           chan *gossipv1.ObservationRequest
+	sendC                  chan []byte
+	signedInC              chan *gossipv1.SignedVAAWithQuorum
+	priv                   p2pcrypto.PrivKey
+	gk                     *ecdsa.PrivateKey
+	gst                    *node_common.GuardianSetState
+	networkID              string
+	bootstrapPeers         string
+	nodeName               string
+	disableHeartbeatVerify bool
+	rootCtxCancel          context.CancelFunc
+	gov                    *governor.ChainGovernor
+	acct                   *accountant.Accountant
+	signedGovCfg           chan *gossipv1.SignedChainGovernorConfig
+	signedGovSt            chan *gossipv1.SignedChainGovernorStatus
+	components             *Components
+}
+
+func NewG(nodeName string) *G {
```

Comment:
As a testing function, you really want to pass `t *testing.T` and run [t.Helper()](https://pkg.go.dev/testing#T.Helper). It makes it so much more useful if there is a failure in that code to not display it.